boards/arc/nsim/nsim_em.yaml

```diff
@@ -2,6 +2,7 @@ identifier: nsim_em
 name: EM Nsim simulator
 type: sim
 simulation: nsim
+simulation_exec: nsimdrv
```

Comment:
Do we really need `simulation_exec`? Should `simulation` not be enough to infer that?

For example, we already have:

https://github.com/zephyrproject-rtos/zephyr/blob/adbbb02525bb178871eda240328d5adfdce71d80/scripts/pylib/twister/twisterlib/testinstance.py#L144-L175

Reply:
prefer to keep those separate, consider the simulation as the class, the executable can be different based on board for example. We have this case already with xtensa simulator, though not upstream.

Reply:
Sure, so does ARM FVP.

It would be nicer if there is a "simulation handler" that resolves the correct executable name based on the board metadata; but, that does seem to be a little convoluted at this time.

Reply:
yes, let me take a look how can this be improved.